sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -845,6 +845,25 @@ static const int ImmediateCommandlistsSetting = [] {
   return std::stoi(ImmediateCommandlistsSettingStr);
 }();
 
+// Get value of the threshold for number of events in immediate command lists.
+// If number of events in the immediate command list exceeds this threshold then
+// cleanup process for those events is executed.
+static const size_t ImmCmdListsEventCleanupThreshold = [] {
+  const char *ImmCmdListsEventCleanupThresholdStr = std::getenv(
+      "SYCL_PI_LEVEL_ZERO_IMMEDIATE_COMMANDLISTS_EVENT_CLEANUP_THRESHOLD");
+  static constexpr int Default = 100;
```

Comment:
What made you choose 100? Can it be lower?

Reply:
It can be lower, what number do you think would be better?

Reply:
Discussed with Rajiv offline, agreed on 20 for now. It can be changed after benchmarking.